From d159461a6977d4d31fcbe73e1ec5bf3aa7b170d0 Mon Sep 17 00:00:00 2001
From: Eliad Peller <eliad@wizery.com>
Date: Mon, 4 Apr 2011 17:56:35 +0300
Subject: [PATCH 27/75] wl12xx: free links on interface removal

TODO: this should be done in a more robustic way during the
the role removal. leave it in a separate patch for now, so
i'll remember doing it :)
---
 drivers/net/wireless/wl12xx/main.c |    3 +++
 1 files changed, 3 insertions(+), 0 deletions(-)

diff --git a/drivers/net/wireless/wl12xx/main.c b/drivers/net/wireless/wl12xx/main.c
index 0e227d4..d6c08db 100644
--- a/drivers/net/wireless/wl12xx/main.c
+++ b/drivers/net/wireless/wl12xx/main.c
@@ -1886,6 +1886,9 @@ static void __wl1271_op_remove_interface(struct wl1271 *wl,
 	wl->ap_fw_ps_map = 0;
 	wl->ap_ps_map = 0;
 	wl->sched_scanning = false;
+	wl->role_id = WL1271_INVALID_ROLE_ID;
+	wl->dev_role_id = WL1271_INVALID_ROLE_ID;
+	wl->system_hlid = WL1271_INVALID_LINK_ID;
 	memset(wl->roles_map, 0, sizeof(wl->roles_map));
 	memset(wl->links_map, 0, sizeof(wl->links_map));
 
-- 
1.7.0.4

